mesh/mesh.go

```diff
@@ -754,11 +767,18 @@ func (msh *Mesh) accumulateRewards(l *types.Layer, params Config) {
 			msh.With().Warning("Atx from block not found in db", log.Err(err), bl.ID(), bl.ATXID)
 			continue
 		}
-		ids = append(ids, atx.Coinbase)
-
+		coinbases = append(coinbases, atx.Coinbase)
+		//create a 2 dimensional map where the entries are
+		//coinbasesAndSmeshers[coinbase_id][smesher_id] = number of blocks this pair has created
+		if _, exists := coinbasesAndSmeshers[atx.Coinbase]; !exists {
+			coinbasesAndSmeshers[atx.Coinbase] = make(map[string]uint64)
+			coinbasesAndSmeshers[atx.Coinbase][atx.NodeID.String()]++
```

Comment:
I still find this counterintuitive. Could you add a comment here explaining why `atx.Coinbase` can be used as a map key but `atx.NodeID` cannot, and why you need to turn it into a string here?